### PR TITLE
Migration guide for flutter/flutter#49389

### PR DESCRIPTION
**Pull request by dnfield**
For what it's worth, I have not been able to find any instances of people overriding `ImageProvider.resolve` in google3 or pub.

This is combined because the upstream change is combined.  Before landing I will change the TBD to actually have the Flutter version it lands in.

### CODE REVIEW

File: src/docs/release/breaking-changes/image-cache-and-provider.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+---
+title: [Changes to ImageCache and ImageProvider]
```
Comment (goderbauer) [inline src/docs/release/breaking-changes/image-cache-and-provider.md]: are these supposed to be in `[]`? The one other guide I checked didn't have that...

File: src/docs/release/breaking-changes/image-cache-and-provider.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+description: [ImageCache requires implementers to override containsKey, and
+ImageProvider has marked resolve as @nonVirtual]
+---
+
+
+## Summary
+
+`ImageCache` now has a method called `obaintsKey`. `ImageProvider` subclasses
+should not override `resolve`, but instead should resolve new methods on
```
Comment (goderbauer) [inline src/docs/release/breaking-changes/image-cache-and-provider.md]: should resolve -> should implement 

or something like that?

File: src/docs/release/breaking-changes/image-cache-and-provider.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    // create stream
+    // set up error handling
+    // interact with ImageCache
+    // call obtainKey/load, etc.
+  }
+
+  ...
+}
+```
+
```
Comment (goderbauer) [inline src/docs/release/breaking-changes/image-cache-and-provider.md]: Is this missing a "Code after the migration:"?

File: src/docs/release/breaking-changes/image-cache-and-provider.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  }
+
+  ...
+}
+
+```
+
+## Timeline
+
+TBD - will update before landing
```
Comment (goderbauer) [inline src/docs/release/breaking-changes/image-cache-and-provider.md]: Reminder to change this :)

File: src/docs/release/breaking-changes/image-cache-and-provider.md (hunk 2 above)
Comment (dnfield) [reply src/docs/release/breaking-changes/image-cache-and-provider.md]: Oops - yeah.

File: src/docs/release/breaking-changes/image-cache-and-provider.md (hunk 3 above)
Comment (dnfield) [reply src/docs/release/breaking-changes/image-cache-and-provider.md]: Done

File: src/docs/release/breaking-changes/image-cache-and-provider.md (hunk 1 above)
Comment (dnfield) [reply src/docs/release/breaking-changes/image-cache-and-provider.md]: Done

File: src/docs/release/breaking-changes/image-cache-and-provider.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
+---
+title: Changes to ImageCache and ImageProvider
+description: ImageCache requires implementers to override containsKey, and
+ImageProvider has marked resolve as @nonVirtual
+---
+
+
+## Summary
+
+`ImageCache` now has a method called `obaintsKey`. `ImageProvider` subclasses
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/image-cache-and-provider.md]: obaintsKey => containsKey. ??

Comment (dnfield) [reply src/docs/release/breaking-changes/image-cache-and-provider.md]: Whups.  Yes.  There is also a method calld `obtainKey`, must've had a brain fart in there haha

**Comment by dnfield** [comment]
Added release info.

**Comment by Gperez88** [comment]
when they move this to stable branch?

**Pull request merged**